jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/RetrieveRegistryCredentialsStep.java

```diff
@@ -82,22 +87,29 @@ static RetrieveRegistryCredentialsStep forTargetImage(
   @Override
   @Nullable
   public Credential call() throws CredentialRetrievalException {
-    buildLogger.lifecycle(String.format(DESCRIPTION, registry) + "...");
+    String description = makeDescription(registry);
+    emit(LogEvent.lifecycle(description + "..."));
 
-    try (Timer ignored = new Timer(buildLogger, String.format(DESCRIPTION, registry))) {
+    try (Timer ignored =
+        LoggingTimer.newTimer(logMessage -> emit(LogEvent.debug(logMessage)), description)) {
       for (CredentialRetriever credentialRetriever : credentialRetrievers) {
         Credential credential = credentialRetriever.retrieve();
         if (credential != null) {
           return credential;
         }
       }
 
-      /*
-       * If no credentials found, give an info (not warning because in most cases, the base image is
-       * public and does not need extra credentials) and return null.
-       */
-      buildLogger.info("No credentials could be retrieved for registry " + registry);
+      // If no credentials found, give an info (not warning because in most cases, the base image is
+      // public and does not need extra credentials) and return null.
+      emit(LogEvent.info("No credentials could be retrieved for registry " + registry));
```

Comment:
IMHO: this should be a `NoCredentialsForRegistryEvent`, and it should include the list of credentialRetrievers requested.  This base code should be totally oblivious to knowing how the results are reported to the user.

Reply:
Oh yes, we will definitely be adding more events with actual structured data for the user to use. For now, these pull requests will first just replace the `JibLogger` code with `LogEvent`s to maintain the same functionality.